### PR TITLE
Forms 7220 Handling impact on AF (v2/core-components) Authoring upon removal of DoR pod from RDE

### PR DESCRIPTION
**Pull request by Yash-099**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -279,6 +279,17 @@
                                                     </cui>
                                                 </uiSettings>
                                             </thankYouMessage>
+                                            <dorAbsenceMessage
+                                            granite:class="warning-icon"
+                                            granite:id="dorWarningIcon"
+                                            jcr:primaryType="nt:unstructured"
+                                            sling:resourceType="/libs/fd/af/components/dorAbsenceBanner"
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml:286]: is it in public area. we cannot use anything in core components that is internal.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -279,6 +279,17 @@
...
+                                            text="Use a non-Rapid Development Environment to use or test a Submit actions that requires the Document of Record feature"
+                                            variant="minimal">
+                                            <granite:rendercondition
+                                                jcr:primaryType="nt:unstructured"
+                                                sling:resourceType="/libs/fd/af/dor/rendercondition/dorAbsence"/>
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml:291]: ditto

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js [hunk 3]
```diff
@@ -38,6 +37,35 @@
...
+
+                dorPreviewElement.removeAttribute("title");
+                var wrapper = document.createElement('span');
+                wrapper.setAttribute("class","dor-preview-button-wrapper");
+                dorPreviewElement.parentNode.insertBefore(wrapper, dorPreviewElement);
+                wrapper.appendChild(dorPreviewElement);
+                var dor_message_tooltip = new Coral.Tooltip();
+                dor_message_tooltip.set({
+                    content: {
+                        innerHTML: "The Document of Record feature is not available for Rapid Development Environment"
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js:57]: i18n missing.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js [hunk 4]
```diff
@@ -38,6 +37,35 @@
...
+        toolBar.addEventListener('DOMNodeInserted',() => {
+            dorPreviewElement = document.querySelector(`[data-action="dorPreview"][icon="printPreview"]`);
+            
+            if (dorPreviewElement && !dorPreviewElement.disabled){
+                dorPreviewElement.disabled = true;
+
+                dorPreviewElement.removeAttribute("title");
+                var wrapper = document.createElement('span');
+                wrapper.setAttribute("class","dor-preview-button-wrapper");
+                dorPreviewElement.parentNode.insertBefore(wrapper, dorPreviewElement);
```
Comment (nit23uec) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js:52]: can you pls share the screen shots also in this review?

**Review by nit23uec — CHANGES_REQUESTED** [review]
inline comments.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js [hunk 5]
```diff
@@ -38,6 +37,35 @@
             };
         };
 
+    var toolBar = document.getElementById("EditableToolbar");
+
+    if(Granite && Granite.isRDE){
```
[the remark this answers is not in the document]
Comment (Yash-099) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js:42]: We have set this in `content/src/main/resources/libs/fd/af/components/RDE/RDE.html` and this component is rendered in `content/src/main/resources/libs/fd/af/content/editors/form-v2/_jcr_content/content/items/rde.xml` based on render condition which checks for rde, thus we get the Granite.isRDE variable in v2 editor when it is RDE. These changes are in cq-guides.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js (hunk 4 above)
Comment (Yash-099) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js:52]: Screenshot 2023-03-28 at 11 43 46 AM
Screenshot 2023-03-28 at 11 41 15 AM

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/css/replacehook.css [hunk 6]
```diff
@@ -13,3 +13,17 @@
 .cmp-replace-dialog-search-components {
     padding: 0;
 }
+
+#EditableToolbar > button:first-child{
```
Comment (suryansh29) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/css/replacehook.css:17]: should use bem

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml (hunk 2 above)
Comment (rismehta) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml:291]: resource type should not start with /libs or /apps, it should be relative to support overlay

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js (hunk 3 above)
Comment (rismehta) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/dorhook.js:57]: +1. Granite.i18n.get

**Comment by rismehta** [comment]
I will fix the coverage part today, looks like it was not reported correctly on my PR

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml (hunk 2 above)
Comment (Yash-099) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml]: Removed /libs

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml (hunk 1 above)
Comment (Yash-099) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/_cq_dialog/.content.xml]: Raised a pr for adding the FinalArea property for the components. https://git.corp.adobe.com/livecycle/cq-guides/pull/5456/files

**Comment by rismehta** [comment]
RDE related deployment information would confuse core component developers, hence closing this PR. Ideally, we need to handle this at API layer

**Pull request closed without merging**